src/devtools/client/debugger/src/components/Editor/LineNumberTooltip.js

```diff
@@ -34,8 +34,12 @@ export function LineNumberTooltip({
     // for the same line number. In that case, we shouldn't run
     // the analysis again.
     if (lineNumber !== lastHoveredLineNumber.current) {
-      runAnalysisOnLine(cx, lineNumber);
       lastHoveredLineNumber.current = lineNumber;
+      setTimeout(() => {
+        if (lineNumber === lastHoveredLineNumber.current) {
+          runAnalysisOnLine(cx, lineNumber);
+        }
+      }, 100);
```

Comment:
No, `throttle` wouldn't be better, it would be worse. `throttle` guarantees the execution of the function regularly and that is something we want to *avoid*. Imagine you move the mouse over 20 lines in the gutter within 1 second (i.e. 50ms per line) and then stop at the last line. `throttle` would request up to 10 analyses whereas my solution will request exactly 1 for the last line (and that's the only one you want).
IMHO `throttle` only makes sense for idempotent functions.

Reply:
that's interesting.  I think i like the idea that the first one will be fast though, so lets see how this feels and perhaps revisit...

agreed that out of the box throttle is not right though